support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,243 @@
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import org.apereo.cas.TicketSerializer;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketCatalog;
+import org.apereo.cas.ticket.TicketDefinition;
+import org.apereo.cas.ticket.TicketState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static org.apereo.cas.config.CassandraTicketRegistryTicketCatalogConfiguration.*;
+
+/**
+ * @author David Rodriguez
+ *
+ * @since 5.2.0
+ */
+public class CassandraTicketRegistry<T> extends AbstractTicketRegistry {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraTicketRegistry.class);
+
+    private static final int FIRST_COLUMN_INDEX = 0;
+    private static final long TEN_SECONDS = 10000L;
+    private static final int TEN = 10;
+    private static final int TICKETS_IN_TESTS = 10;
+
+    private final TicketCatalog ticketCatalog;
+    private final TicketSerializer<T> serializer;
+    private final Class<T> typeToWriteToCassandra;
+
+    private final PreparedStatement insertTgtStmt;
+    private final PreparedStatement updateTgtStmt;
+    private final PreparedStatement selectTgtStmt;
+    private final PreparedStatement deleteTgtStmt;
+
+    private final PreparedStatement insertStStmt;
+    private final PreparedStatement updateStStmt;
+    private final PreparedStatement selectStStmt;
+    private final PreparedStatement deleteStStmt;
+
+    private final PreparedStatement selectExStmt;
+    private final PreparedStatement selectDateExStmt;
+
+    private final PreparedStatement selectLrStmt;
+    private final PreparedStatement updateLrStmt;
+
+    private final Session session;
+
+    public CassandraTicketRegistry(final TicketCatalog ticketCatalog, final String contactPoints, final String username, final String password,
+                                   final String keyspace, final TicketSerializer<T> serializer, final Class<T> typeToWriteToCassandra) {
+        this.ticketCatalog = ticketCatalog;
+        this.serializer = serializer;
+        this.typeToWriteToCassandra = typeToWriteToCassandra;
+        final Cluster cluster = Cluster.builder().addContactPoints(contactPoints.split(",")).withCredentials(username, password)
+                .withProtocolVersion(ProtocolVersion.V3).build();
+
+        this.session = cluster.connect(keyspace);
+
+        this.selectTgtStmt = session.prepare("select ticket from " + TGT_TABLE + " where id = ?").setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM);
```

Comment:
Things like this need to be removed. You want to look up table names from the catalog. Queries and statements need to be built on the fly and then cached if needed on demand.